cloud-checksum/src/cli.rs

```diff
@@ -456,6 +482,10 @@ pub struct Copy {
     pub metadata_mode: MetadataCopy,
     #[arg(long, env, default_value = "server-side")]
     pub copy_mode: CopyMode,
+
+    #[arg(long, env, default_value = "default-environment")]
+    pub destination_credential_provider: CredentialProvider,
+
```

Comment:
I will add a separate issue for this feature - but rclone allows us to use one IAM user for source and one for destination (say). This sometimes is useful copying between different organisations who insist on issuing IAM users rather than using AWS permissions.

Reply:
This would then be restricted to a download/upload copy - server side copy will only work if both source and destination credentials are identical.